packages/tools/pie-icons-react/scripts/react-icons-gen.js

```diff
@@ -14,34 +14,57 @@ async function checkDirExists (directoryPath) {
         await fs.ensureDir(directoryPath);
         console.log(`Directory "${directoryPath}" exists.`);
     } catch (err) {
-        console.error(err)
+        console.error(err);
     }
 }
 
-const handleComponentName = name => name.replace(/\-(\d+)/, '$1'); // eslint-disable-line no-useless-escape
+const handleComponentName = (name) => name.replace(/\-(\d+)/, '$1'); // eslint-disable-line no-useless-escape
 
 // check that the /icons directory exists, if not create it
 checkDirExists(ICONS_DIR);
 
 // open a write stream to index.tsx
 const indexFile = fs.createWriteStream(
     filePath,
-    err => {
+    (err) => {
         console.error(err);
     }
 );
 
+const template = (variables, { tpl }) => {
+    return tpl`
+    ${variables.imports};
+    ${variables.interfaces};
+    const ${variables.componentName} = (props: any) => {
+        const { className, ...remainingProps } = props;
+
+        return (${variables.jsx});
+    }
+
+    ${variables.exports};
+`;
+}
+
 // loop through the icons in pie-icons, generate each component and add it to the index.tsx
-Object.keys(icons).map(iconKey => {
-    const pathPrefix = icons[iconKey].pathPrefix;
+Object.keys(icons).map((iconKey) => {
+    const { pathPrefix } = icons[iconKey];
     const capitalisedPathPrefix = (pathPrefix !== undefined ? (pathPrefix).substring(1, 2).toUpperCase() + (pathPrefix).substring(2) : '');
     const componentName = `Icon${capitalisedPathPrefix + pascalCase(handleComponentName(iconKey))}`;
-    const Comp = svgr.transform.sync(
+    let Comp = svgr.transform.sync(
         icons[iconKey].toSvg(),
-        { icon: true, typescript: true },
-        { componentName }
+        {
+            icon: true,
+            typescript: true,
+            template: template,
+        },
+        { componentName },
     );
-    fs.writeFile(`${ICONS_DIR}/${componentName}.tsx`, Comp, err => {
+
+    const customClassName = `className={"pie-icon pie-icon--${iconKey.toLowerCase()}" + (props.className ? ' ' : '') + (props.className ?? '') }`;
```

Comment:
This should work too:

```suggestion
    const customClassName = `className={"pie-icon pie-icon--${iconKey.toLowerCase()}" + (props.className ? '${props.className} ' : '') }`;
```